### ISSUE TITLE
Re-add widgets to their previous locations?

### ISSUE THREAD

**Issue by mj-sson** [zobnin/aio#87]
When you add widgets, they are all inserted at the top of the home screen. This is fine when adding a widget for the first time. But when you remove a widget and add it a second time, I think it would be more convenient if that widget was re-added to the same location it was originally removed from.

This would be really helpful when you have a lot of widgets on the home screen and you need to remove some of them temporarily. Now, everytime you re-add the widgets, they wind up at the top of the screen, meaning you have to re-sort them all over again.

In a way, this could also be seen as a complement to the widget switcher:

Now, you can temporarily SWITCH ON the widgets you need, and they will appear unsorted at the top of the screen, until you switch them off again.

Later, you'd also be able to temporarily SWITCH OFF some of the widgets on the home screen, until you switch them back on again, re-appearing where they were before.

This surely isn't a new idea, just something I noticed when trying out the widget switcher and my widgets got all rearranged.

**Comment by zobnin** [comment]
There are several reasons why it's not done that way:

1. Widgets have no fixed position on the screen. They are all positioned relative to each other. This is not such a big problem since a widget can be added to "about the same" position. But still.

2. Forgetting problem. Let's say you have been using a Control Panel widget for some time, which you had at the very bottom of the screen. Then you realized that you don't need it. And after some time you decided to temporarily add it to the screen. It would be wrong to add it to the end of the screen. That is, you need some period of forgetting the position. A week, for example. However, in this case, for users it will look as if the widget is added to random places on the screen for some unknown reason.

PS You can solve your problem right now with the help of scripts. They allow you to find out the current position of the widget and add it to "approximately the same" position. Moreover, in my script repository there is even a script that allows you to save and restore the entire screen state with all widgets and their positions. This can be used for e.g. implementing profiles, when you have several screen configurations that you can switch between. 

TLDR; further thought and study of the issue is required

**Comment by mjdjohansson** [comment]
Thank you for the clarification! Yes, I can see how that complicates things. 

I tried the Screen state sample script you suggested and I really like the idea of having a quick way to backup and restore the screen layout. I don't think I would have found it if you hadn't told me about it.

This version seems to reorder some of my widgets and create a new empty apps list every time when restoring, but I'm really impressed with all the hidden treasures the AIO Launcher keeps offering!

Maybe an official version of this script could be added to the AIO Store in the future? 

Anyhow, thanks again for your help :)

**Comment by mjdjohansson** [comment]
I noticed the Plugins link to https://execbit.ru/aiolauncher/plugins.html on https://github.com/zobnin/aio doesn't seem to work.

**Comment by zobnin** [comment]
> I noticed the Plugins link to https://execbit.ru/aiolauncher/plugins.html on https://github.com/zobnin/aio doesn't seem to work.

Fixed. Thanks.

**Comment by zobnin** [comment]
> Maybe an official version of this script could be added to the AIO Store in the future?

Maybe. Or it can be integrated in the launcher itself.

**Comment by mjdjohansson** [comment]
Even better 😉

**Comment by mj-sson** [comment]
Solved with 'Profiles' in 5.4.0 🙂
